### PR TITLE
Image & Image Alias lifecycle events

### PR DESCRIPTION
**Pull request by masnax**
* Created, Deleted, Updated, Retrieved, Pushed, Refreshed, and Secret Created events for Images
* CRUD for Image aliases

* I noticed `imageAliasDeleted` was missing its swagger documentation, so I added it.

### CODE REVIEW

File: lxd/images.go [hunk 1]
```diff
@@ -3455,6 +3455,8 @@ func imageExportPost(d *Daemon, r *http.Request) response.Response {
 			return fmt.Errorf(opWaitAPI.Err)
 		}
 
+		d.State().Events.SendLifecycle(projectName, lifecycle.ImagePushed.Event(fingerprint, projectName, op.Requestor(), log.Ctx{"target": req.Target}))
```
Comment (stgraber) [inline lxd/images.go:3482]: I think this should use `ImageRetrieved` as it's really the same from an audit point of view (the full image data was transferred out).

File: lxd/images.go [hunk 2]
```diff
@@ -3607,6 +3607,9 @@ func imageRefresh(d *Daemon, r *http.Request) response.Response {
 	// Begin background operation
 	run := func(op *operations.Operation) error {
 		_, err := autoUpdateImage(d.ctx, d, op, imageId, imageInfo, projectName, true)
+		if err != nil {
+			d.State().Events.SendLifecycle(projectName, lifecycle.ImageRefreshed.Event(fingerprint, projectName, op.Requestor(), nil))
```
Comment (stgraber) [inline lxd/images.go:3635]: I wonder if `autoUpdateImage` returns us something to figure out whether a refresh actually happened, if so, we should only issue the event if an update was actually found.

Comment (stgraber) [reply lxd/images.go:3635]: I believe we also want to issue this event for the scheduled image refreshes.

File: lxd/images.go [hunk 3]
```diff
@@ -2264,6 +2264,8 @@ func imageDelete(d *Daemon, r *http.Request) response.Response {
 		// Remove main image file from disk.
 		imageDeleteFromDisk(imgInfo.Fingerprint)
 
+		d.State().Events.SendLifecycle(projectName, lifecycle.ImageDeleted.Event(imgInfo.Fingerprint, projectName, op.Requestor(), nil))
```
Comment (stgraber) [inline lxd/images.go:2268]: I believe we should also issue this event when an image is automatically deleted.

File: lxd/images.go [hunk 4]
```diff
@@ -1017,6 +1018,8 @@ func imagesPost(d *Daemon, r *http.Request) response.Response {
 			return errors.Wrapf(err, "Failed syncing image between nodes")
 		}
 
+		d.State().Events.SendLifecycle(projectName, lifecycle.ImageCreated.Event(info.Fingerprint, projectName, op.Requestor(), log.Ctx{"type": info.Type}))
```
Comment (stgraber) [inline lxd/images.go:1022]: We should make sure that when LXD automatically downloads an image (as happens during `lxc launch`) that we also issue this event upon initial image download.

Comment (masnax) [reply lxd/images.go:1022]: Just added that event in `ImageDownload` under `lxd/daemon_images.go`

File: lxd/images.go (hunk 3 above)
Comment (masnax) [reply lxd/images.go:2275]: Added it to `pruneExpiredImagesInProject`, and passed around the operation for the requestor.

File: lxd/images.go (hunk 2 above)
Comment (masnax) [reply lxd/images.go]: I just moved the lifecycle event into the result check in `autoUpdateImage`.

**Pull request merged**